src/components/accession2/viabilityTesting/ViabilityResultModal.tsx

```diff
@@ -0,0 +1,65 @@
+import { Box, Typography } from '@mui/material';
+import { Button, DialogBox } from '@terraware/web-components';
+import { Accession2, updateAccession2 } from 'src/api/accessions2/accession';
+import strings from 'src/strings';
+import useSnackbar from 'src/utils/useSnackbar';
+import { ViabilityTest } from 'src/api/types/accessions';
+
+export interface ViabilityResultModalProps {
+  open: boolean;
+  accession: Accession2;
+  onClose: () => void;
+  reload: () => void;
+  viabilityTest: ViabilityTest;
+}
+
+export default function ViabilityResultModal(props: ViabilityResultModalProps): JSX.Element {
+  const { onClose, open, accession, reload, viabilityTest } = props;
+
+  const snackbar = useSnackbar();
+
+  const saveResult = async () => {
+    if (accession) {
+      const newAccession = { ...accession, viabilityPercent: getViabilityResult() };
+      const response = await updateAccession2(newAccession);
+
+      if (response.requestSucceeded) {
+        reload();
+        onClose();
+      } else {
+        snackbar.toastError();
+      }
+    }
+  };
+
+  const getViabilityResult = () => {
+    let sum = 0;
+    viabilityTest?.testResults?.forEach((tr) => {
+      sum = sum + Number(tr.seedsGerminated);
+    });
+    const result = (sum / Number(viabilityTest.seedsTested)) * 100;
```

Comment:
can we use the percent in the viability test returned from BE? it's ok to reload data.

Reply:
I tried to do that, but the problem is that if we are creating a new test we don't have the id to get that value back from the backend

Reply:
hmm, ok got it, we have no way of back referencing the viability test since BE returns the entire accession. maybe we can bring this up with Steve.. for now this is ok maybe.